File: ejercicios/AMarotoChivite/AMarotoChivite-01-10-Catan-Java/src/test/java/gameCatan/utils/randomDataTest.java
package gameCatan.utils;

import gameCatan.enums.TypeResource;
import org.junit.jupiter.api.Test;

import static gameCatan.utils.randomData.randomType;
import static org.junit.jupiter.api.Assertions.fail;

class randomDataTest {

    @Test
    void randomTypeTest() {
        TypeResource typeTest = randomType();
        TypeResource[] correctTypes = {TypeResource.WOOD, TypeResource.COAL, TypeResource.SEED};

        int count = 0;
        for (int i = 0; i < correctTypes.length; i++) {
            if (correctTypes[i].equals(typeTest)) {
                count++;
            }
        }
        if (count != 1) {
            fail();
        }
    }
}

File: ejercicios/AMarotoChivite/AMarotoChivite-01-10-Catan-Kotlin/src/test/kotlin/gameCatan/utils/randomDataTest.kt
package gameCatan.utils

import gameCatan.enums.TypeResource
import gameCatan.utils.randomData.randomType
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class randomDataTest {
    @Test
    fun randomTypeTest() {
        val typeTest = randomType()
        val correctTypes = arrayOf(TypeResource.WOOD, TypeResource.COAL, TypeResource.SEED)
        var count = 0
        for (i in correctTypes.indices) {
            if (correctTypes[i] == typeTest) {
                count++
            }
        }
        if (count != 1) {
            Assertions.fail<Any>()
        }
    }
}